source/addonHandler/__init__.py

```diff
@@ -751,17 +751,23 @@ def getCodeAddon(obj=None, frameDist=1):
 	# Not found!
 	raise AddonError("Code does not belong to an addon")
 
+
+translationFunctions = ["_", "ngettext", "pgettext", "npgettext"]
+
+
 def initTranslation():
 	addon = getCodeAddon(frameDist=2)
 	translations = addon.getTranslationsInstance()
 	# Point _ to the translation object in the globals namespace of the caller frame
-	# FIXME: should we retrieve the caller module object explicitly?
 	try:
 		callerFrame = inspect.currentframe().f_back
-		callerFrame.f_globals['_'] = translations.gettext
-		# Install pgettext and npgettext function.
-		callerFrame.f_globals['pgettext'] = translations.pgettext
-		callerFrame.f_globals['npgettext'] = translations.npgettext
+		module = inspect.getmodule(callerFrame)
+		[

```

Comment:
Using list comprehension in cases where a simple loop would suffice and you do not filter the original list in any way is generally regarded as a code smell. Also this mapping from `gettext` to `_` could be made more explicit by creating a dictionary from translation functions to the names in which they're going to be installed i.e. from `translation.gettext` to `_`, iterating through it and setting them as an attribute of the calling module.

Reply:
I personally don't see a problem in list comprehension here, as it removes code repetition in current implementation. As for dictionary, i don't see a point here, because _ is the only function that should be installed differently and three other functions, which keys and values are equal, what is the point here?

Reply:
Using list comprehension here does not remove code repetition in any way, it also forces you to write more lines of code and increase indentation needlessly.
Compare the current implementation with the following:
```python
for func in translationFunctions:
    setattr(...)
```
IMHO much shorter and nicer.
As to the dictionary it brings two advantages:
- clearer intent - you can do `for transFunc, installAs in dict.items()` explicit names are always better
- we needed to install `gettext` under a different name such case may occur again, for a different function and the current implementation does not give necessary flexibility to cover this

Reply:
Ok, listening for pros and cons, i will do it to minimize future impact, however i cannot think of the case when we will need to change something here.
Anyway, i'll do it.

Reply:
What do you think with approach?

_TRANSLATION_FUNCTIONS = {"gettext": "_", "ngettext": "", "pgettext": "", "npgettext": ""}
		for funcName, installAs in _TRANSLATION_FUNCTIONS.items():
			func = getattr(translations, funcName)
			setattr(module, installAs if installAs else funcName, func)

@lukaszgo1

Reply:
I'd personally avoid the `getattr` call by storing the functions in the dictionary, and use explicit names for consistency as follows:
```python
_TRANSLATION_FUNCTIONS = {translations.gettext: "_", translations.ngettext: "ngettext", translations.pgettext: "pgettext", translations.npgettext: "npgettext"} 
```